### DISCUSSION TITLE
Adding parameter - ROW_FORMAT = COMPRESSED for table: raw-2021-10-27

### DISCUSSION THREAD

**Discussion by alexsoftum** [postalserver/postal#1664] [Feature suggestions]
Hello!
When sending large letters, the table - raw-2021-10-27 starts to take up a lot of disk space. It would be great if the - ROW_FORMAT = COMPRESSED parameter was added by default when creating a table. This will allow the database to compress data on the fly and store it compactly in the database.

As sample:

CREATE TABLE IF NOT EXISTS `raw-2021-10-27` (
`id` int NOT NULL AUTO_INCREMENT PRIMARY KEY,
`data` longblob,
`next` int DEFAULT NULL
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 ROW_FORMAT=COMPRESSED;

To check the space occupied by a table, you can use the SQL query: SELECT data_length AS size 
FROM information_schema.TABLES 
WHERE TABLE_SCHEMA = 'mdb-server-1' 
AND TABLE_NAME = 'raw-2021-10-27';

All tables:
SELECT table_name AS "Table", ROUND (((data_length + index_length) / 1024/1024), 2) AS "Size in (MB)" FROM information_schema.TABLES WHERE TABLE_SCHEMA = ' mdb-server-1 'ORDER BY (data_length + index_length) DESC;

**Comment by willpower232** [comment]
I can imagine there might be a performance penalty in CPU usage somewhere along the line so it might be a problem if you are using a less well provisioned server. Postal can regulate storage usage itself if you like. As discussed in #1648 Postal shouldn't be storing messages long term and the content should be mostly stored by the application sending the messages.

**Reply by alexsoftum** [reply]
Thanks for the answer. 

I did not find any noticeable decrease in performance. But the result was shocking! Instead of 18GB of occupied space - 153MB. The compression is just incredible on the same type of mailings! In addition, when sending letters, the counter of the size of the table used is constantly updated. This is not reasonable considering that MySQL calculates it by itself. The size of the table is easy to get from the database - information_schema at any time without additional inserts and updates of records in the database.

I suggest adding an option to the postal.yaml configuration to enable DB table compression.
